src/pluto/operation/pad.cr

```diff
@@ -0,0 +1,54 @@
+module Pluto::Operation::Pad
+  def pad(all : Int32 = 0, *, left : Int32 = 0, right : Int32 = 0, top : Int32 = 0, bottom : Int32 = 0, pad_type : PadType = PadType::Black) : self
+    clone.pad!(all, left: left, right: right, top: top, bottom: bottom, pad_type: pad_type)
+  end
+
+  def pad!(all : Int32 = 0, *, left : Int32 = 0, right : Int32 = 0, top : Int32 = 0, bottom : Int32 = 0, pad_type : PadType = PadType::Black) : self
+    top = top > 0 ? top : all
+    bottom = bottom > 0 ? bottom : all
+    left = left > 0 ? left : all
+    right = right > 0 ? right : all
+
+    new_width = left + width + right
+    new_height = top + height + bottom
+
+    each_channel do |channel, channel_type|
+      new_channel = Array(UInt8).new((top + height + bottom) * new_width) do |i|
+        next 0u8 if pad_type.black?
+
+        adjusted_y = (i // new_width - top).clamp(0, height - 1)
+
+        (i % new_width) <= left ? channel[adjusted_y * width] : channel[adjusted_y * width + width - 1]
```

Comment:
We use `#unsafe_fetch` and `#unsafe_put` in other operations, so I think it would be consistent to use them here as well.

Reply:
I refactored this a tad in my next commit, move the `black?` check out of this block so it's not checked for every pixel value.